resources/vep/plugins/AlphScore.pm

```diff
@@ -0,0 +1,123 @@
+package AlphScore;
+
+use strict;
+use warnings;
+use List::Util qw(max);
+
+use Bio::EnsEMBL::Utils::Sequence qw(reverse_comp);
+use Bio::EnsEMBL::Variation::Utils::Sequence qw(get_matched_variant_alleles);
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+use Bio::EnsEMBL::Variation::VariationFeature;
+
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+=head1 NAME
+ AlphScore
+=head1 SYNOPSIS
+ mv AlphScore.pm ~/.vep/Plugins
+ ./vep -i variations.vcf --plugin AlphScore,/FULL_PATH_TO_ALPHSCORE_file
+=head1 DESCRIPTION
+ Plugin to annotate AlphScore scores as described in https://academic.oup.com/bioinformatics/article/39/5/btad280/7135835
+=cut
+
+my $output_vcf;
+
+my $self;
+
+sub new {
+    if (!(defined $self)) {
+        my $class = shift;
+        $self = $class->SUPER::new(@_);
+
+        $self->expand_left(0);
+        $self->expand_right(0);
+
+        my $score_file = $self->params->[0];
+        my $mappingFile = $self->params->[1];
+        die("ERROR: input file not specified\n") unless $score_file;
+        my %gene_mapping = parseMappingFile($mappingFile);
+
+        $self->add_file($score_file);        
+        $self->{gene_mapping} = \%gene_mapping;
+    }
+  return $self;
+}
+
+sub feature_types {
+  return ['Transcript'];
+}
+
+sub get_header_info {
+    $self = AlphScore->new;
+
+    my $result;
+    $result->{ALPHSCORE_FILE} = "file:" . $self->params->[0] . "'";
+    $result->{ALPHSCORE} = "AlphScore_final scores as described in https://academic.oup.com/bioinformatics/article/39/5/btad280/7135835";
+    return $result;
+}
+
+sub parseMappingFile {
+    my %mapping_data;
+    my $file = $_[0];
+    open(MAPPING_FH, '<', $file) or die $!;
+
+    my @split;
+    while (<MAPPING_FH>) {
+        my $line = $_;
+        chomp($line);
+        @split = split(/\t/, $line);
+        #approved symbol
+        if (defined $split[0] and length $split[0] and defined $split[3] and length $split[3]){
+            $mapping_data{$split[0]} = $split[3];
+        }
+        #previous symbol
+        if (defined $split[1] and length $split[1] and defined $split[3] and length $split[3]){
+            $mapping_data{$split[1]} = $split[3];
+        }
+        #alias symbol
```

Comment:
should we do something with withdrawn gene symbols, e.g. ZWINTAS?

Reply:
They are not offered by the https://biomart.genenames.org/, I think it would also be fine to not annotate scores from withdrawn genes.